frontend/packages/dev-console/src/components/import/DeployImageForm.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form, Button } from 'patternfly-react';
+import { FormikProps, FormikValues } from 'formik';
+import { ButtonBar } from '@console/internal/components/utils';
+import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+import DeployImageSection from './deploy-image/DeployImageSection';
+import AppSection from './app/AppSection';
+import AdvancedSection from './advanced/AdvancedSection';
+
+export interface DeployImageFormProps {
+  builderImages?: NormalizedBuilderImages;
+}
+
+const DeployImageForm: React.FC<FormikProps<FormikValues> & DeployImageFormProps> = ({
+  values,
+  errors,
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+  dirty,
+}) => (
+  <Form className="co-deploy-image">
+    <div className="co-m-pane__form">
+      <AppSection project={values.project} />
+      <DeployImageSection />
+      <AdvancedSection values={values} />
+    </div>
+    <br />
+    <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
+      <Button disabled={!dirty || !_.isEmpty(errors)} bsStyle="primary" onClick={handleSubmit}>
+        Deploy
+      </Button>
+      <Button onClick={handleReset}>Cancel</Button>
```

Comment:
See how `GitImportForm` handles reset. We should be consistent.
Either it changes or this changes.
Same for submit.

Reply:
This change was @rohitkrai03 idea. We both agree that it is a hack and don't want to keep it but don't have a better solution at the moment. 
Formik does not correctly handle event.preventDefaults and we couldn't shove it in so that it would work.

Reply:
@christianvogt The problem here was the search input needs to trigger search on enter and default behaviour of a form is that it submits if all fields are valid. So, when user wants to search an image it would trigger submit which was not needed. This is why i suggested this fix yesterday. I dig into it a bit and found a solution for it and it works as expected -

```
<Form
  className="co-deploy-image"
  onSubmit={handleSubmit}
  onReset={handleReset}
  onKeyPress={(event: KeyboardEvent) => {
    if (event.which === 13 /* Enter */) {
      event.preventDefault();
    }
  }}
>
  <div className="co-m-pane__form">
    <AppSection project={values.project} />
    <DeployImageSection />
    <AdvancedSection values={values} />
  </div>
  <br />
  <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
    <Button type="submit" disabled={!dirty || !_.isEmpty(errors)} bsStyle="primary">
      Deploy
    </Button>
    <Button type="reset">Cancel</Button>
  </ButtonBar>
</Form>
```

Reply:
Interesting. Thanks @rohitkrai03

Reply:
@joshuawilson @rohitkrai03 wouldn't it be better to stop propagation of the event at the search input level so that the enter key never submits the form?